### PR TITLE
[#IP-182] Upgrade GetMessage model with payee

### PR DESCRIPTION
**Pull request by AleDore**
#### List of Changes

- Upgrade `io-functions-commons` to latest version
- Update Openapi specs
- Add `payee` management to GetMessage handler
- Tests refactor
#### Motivation and Context

This is needed in order to expose an updated model on GetMessage endpoint. In addition this must handle and set payee for payment messages that don't have payee by default (namely old payment messages)
#### How Has This Been Tested?

unit test
#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: GetMessage/handler.ts [hunk 1]
```diff
@@ -61,13 +66,61 @@ type IGetMessageHandler = (
...
+    !PaymentDataWithRequiredPayee.is(maybePaymentData.value)
+  ) {
+    const errorOrMaybeSenderService = await serviceModel.findLastVersionByModelId(
+      [senderServiceId]
+    )();
+    if (E.isLeft(errorOrMaybeSenderService)) {
+      context.log.error(
+        `GetMessageHandler|${JSON.stringify(errorOrMaybeSenderService.left)}`
+      );
+      return E.left<IResponseErrorInternal, O.Option<PaymentData>>(
```
Comment (gquadrati) [inline GetMessage/handler.ts:86]: There is no need of defying types

File: GetMessage/handler.ts [hunk 2]
```diff
@@ -61,13 +66,61 @@ type IGetMessageHandler = (
...
+          `Cannot get message Sender Service|ERROR=${
+            E.toError(errorOrMaybeSenderService.left).message
+          }`
+        )
+      );
+    }
+    const maybeSenderService = errorOrMaybeSenderService.right;
+    if (O.isNone(maybeSenderService)) {
+      // the message sender service does not exist
+      return E.left<IResponseErrorInternal, O.Option<PaymentData>>(
```
Comment (gquadrati) [inline GetMessage/handler.ts:97]: ditto

File: GetMessage/handler.ts [hunk 3]
```diff
@@ -61,13 +66,61 @@ type IGetMessageHandler = (
...
+        ResponseErrorInternal(
+          `Message Sender not found The message that you requested does not have a related sender service`
+        )
+      );
+    }
+    return E.right<IResponseErrorInternal, O.Option<PaymentData>>(
```
Comment (gquadrati) [inline GetMessage/handler.ts:103]: ditto

File: GetMessage/handler.ts [hunk 4]
```diff
@@ -61,13 +66,61 @@ type IGetMessageHandler = (
...
+      O.some({
+        ...maybePaymentData.value,
+        payee: {
+          fiscal_code: maybeSenderService.value.organizationFiscalCode
+        }
+      })
+    );
+  }
+  return E.right<IResponseErrorInternal, O.Option<PaymentData>>(
```
Comment (gquadrati) [inline GetMessage/handler.ts:112]: ditto

File: GetMessage/handler.ts [hunk 5]
```diff
@@ -61,13 +66,61 @@ type IGetMessageHandler = (
   | IResponseErrorInternal
 >;
 
+const getErrorOrPaymentData = async (
```
Comment (gquadrati) [inline GetMessage/handler.ts:69]: Please, add a brief description.

**Review by gquadrati — CHANGES_REQUESTED** [review]
LGTM, just a few comments. Also, did you try it on io-mock env?

**Review by michaeldisaro — APPROVED** [review]
Apart some non functional code, that's used pretty commonly, and @gquadrati notes, I think it's ok.

File: GetMessage/handler.ts [hunk 6]
```diff
@@ -61,13 +66,61 @@ type IGetMessageHandler = (
...
+  context: Context,
+  serviceModel: ServiceModel,
+  senderServiceId: ServiceId,
+  maybePaymentData: O.Option<PaymentData>
+): Promise<E.Either<IResponseErrorInternal, O.Option<PaymentData>>> => {
+  if (
```
[the remark this answers is not in the document]
Comment (gquadrati) [reply GetMessage/handler.ts:75]: I noticed that too, I thought it was for not mixing old imperative code with functional one.

[the remark this answers is not in the document]
Comment (michaeldisaro) [reply GetMessage/handler.ts:75]: I think that the new code should me as more functional as possible... but by the way I don't think it's a reason to block a PR if the code maintains a style that's coherent with other code. I would have requested changes if the code would have changed style.

[the remark this answers is not in the document]
Comment (gquadrati) [reply GetMessage/handler.ts:75]: I agree

File: GetMessage/__tests__/handler.test.ts [hunk 7]
```diff
@@ -189,4 +222,113 @@ describe("GetMessageHandler", () => {
 
     expect(result.kind).toBe("IResponseErrorNotFound");
   });
+
+  it("should respond with a message payment data overriden with payee if original content does not have a payee", async () => {
```
Comment (gquadrati) [inline GetMessage/__tests__/handler.test.ts:226]: Did we test the case when a payee is present?

**Review by gquadrati — APPROVED** [review]
LGTM

**Pull request merged**